content/posts/copr-srpms.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "Building SRPMs in Copr"
+date: 2022-03-07T12:57:40+01:00
+---
+
+### Introduction
+
+If you use Packit to build RPMs for your upstream code changes, likely,
+you have already read about how does Packit build your SRPMs. If not, then just a short recap:
+Each time an RPM build is triggered, Packit builds an SRPM and then submits
+the created SRPM file to Copr where Copr takes care of building the actual RPMs.
+Since users can modify the behaviour of building SRPMs by defining [actions](https://packit.dev/docs/actions/),
+this process needs to be run in an isolated environment. For this, we implemented our
+[sandboxing mechanism](https://github.com/packit/sandcastle),
+which simply runs the provided commands in an Openshift pod freshly created for each build.
+
+#### Problems of the previous workflow for SRPM builds
+
+This is a pretty good-functioning workflow, but it has some downsides which have become more and more annoying
+with the growing user base. Because of the resources, we have set limits for Openshift pods running at one time.
+This directly affects how many SRPM builds can run in parallel. As a result, when there are too many requests for (S)RPM builds,
+some can get stuck in the queue while waiting for other builds to finish. Another inconvenience coming with pods being
+always freshly created is copying the needed data into
+and from the pod. This has also cost us some hours of desperate debugging of weird errors.
+Another disadvantage is that users cannot easily configure dependencies for their actions run during building SRPMs.
+We try to install the dependencies manually on-demand, but of course, this isn't that flexible.
+
+We have been thinking about improving the process for a long time but have never had any clear conclusion.
```

Comment:
This is no longer true (we now have the conclusion) so it can't be present perfect tense.
```suggestion
We were thinking about improving the process for a long time but never reached any clear conclusion.
```